numba/tests/test_ir_inlining.py

```diff
@@ -37,17 +44,41 @@ class InlineTestPipeline(CompilerBase):
     metadata store"""
 
     def define_pipelines(self):
-        pipeline = DefaultPassBuilder.define_nopython_pipeline(
-            self.state, "inliner_custom_pipe")
-        # mangle the default pipeline and inject DCE and IR preservation ahead
-        # of legalisation
-
-        # TODO: add a way to not do this! un-finalizing is not a good idea
-        pipeline._finalized = False
-        pipeline.add_pass_after(PreserveIR, IRLegalization)
-
-        pipeline.finalize()
-        return [pipeline]
+        name = "inliner_custom_pipe"
+        state = self.state
+
+        dpb = DefaultPassBuilder
+        pm = PassManager(name)
+
+        pm = PassManager("inline_tests")
+        pm.add_pass(TranslateByteCode, "analyzing bytecode")
+        pm.add_pass(FixupArgs, "fix up args")
+        pm.add_pass(IRProcessing, "processing IR")
+        pm.add_pass(WithLifting, "Handle with contexts")
+        pm.add_pass(InlineClosureLikes,
+                    "inline calls to locally defined closures")
+        pm.add_pass(RewriteSemanticConstants, "rewrite semantic constants")
+        pm.add_pass(DeadBranchPrune, "dead branch pruning")
+        pm.add_pass(GenericRewrites, "nopython rewrites")
+        pm.add_pass(MakeFunctionToJitFunction,
+                    "convert make_function into JIT functions")
+        pm.add_pass(InlineInlinables, "inline inlinable functions")
+        pm.add_pass(DeadBranchPrune, "dead branch pruning")
+        pm.add_pass(FindLiterallyCalls, "find literally calls")
+        pm.add_pass(LiteralUnroll, "handles literal_unroll")
+        pm.add_pass(ReconstructSSA, "ssa")
+        pm.finalize()
+
+        typed_passes = dpb.define_typed_pipeline(state)
+        pm.passes.extend(typed_passes.passes)
+
+        lowering_passes = dpb.define_nopython_lowering_pipeline(state)
+        pm.passes.extend(lowering_passes.passes)
+
+        pm.add_pass_after(PreserveIR, IRLegalization)
+
+        pm.finalize()
```

Comment:
What's the purpose of this change?

Reply:
I am changing the default nopython pipeline to include the partial constant propagation pass. Which optimizes the IR and breaks existing tests that expect the IR to have certain instructions/certain shape (i.e., `test_inlining_models`).